src/as_execution/abi.rs

```diff
@@ -1531,6 +1531,149 @@ pub(crate) fn assembly_script_chain_id(mut ctx: FunctionEnvMut<ASEnv>) -> ABIRes
     Ok(chain_id as u64)
 }
 
+/// Return the price in nMAS to book an deferred call space in a specific slot.
+#[named]
+pub(crate) fn assembly_script_get_deferred_call_quote(
+    mut ctx: FunctionEnvMut<ASEnv>,
+    deferred_call_period: i64,
+    deferred_call_thread: i32,
+    max_gas: i64,
+) -> ABIResult<u64> {
+    let env = get_env(&ctx)?;
+    sub_remaining_gas_abi(&env, &mut ctx, function_name!())?;
+    let asc_slot: (u64, u8) = match (
+        deferred_call_period.try_into(),
+        deferred_call_thread.try_into(),
+    ) {
+        (Ok(p), Ok(t)) => (p, t),
+        (Err(_), _) => abi_bail!("negative validity end period"),
+        (_, Err(_)) => abi_bail!("invalid validity end thread"),
+    };
+    if max_gas.is_negative() {
```

Comment:
best practice: better do a try_into u64 directly since below we convert using "as u64"

Reply:
done